dapp/src/components/page/proposal/ProposalPage.tsx

```diff
@@ -0,0 +1,130 @@
+  const [voteType, setVoteType] = useState<VoteType>();
+  const [voteStatus, setVoteStatus] = useState<VoteStatus>();
+  const [proposalStatus, setProposalStatus] = useState<ProposalStatus | null>(
+    null,
+  );
+  const [endDate, setEndDate] = useState<string | null>(null);
+
+  const openVotersModal = (voteType: VoteType) => {
+    if (proposalStatus !== "active") {
+      setVoteType(voteType);
+      setIsVotersModalOpen(true);
+    } else {
+      alert("Cannot show voters while voting is in progress");
+    }
+  };
+
+  const openVotingModal = () => {
+    if (proposalStatus === "active") {
+      if (connectedAddress) {
+        setIsVotingModalOpen(true);
+      } else {
+        alert("Please connect your wallet first");
+      }
+    } else {
+      alert("Cannot vote anymore, voting is ended.");
+    }
+  };
+
+  const openExecuteProposalModal = () => {
+    if (proposalStatus === "voted") {
+      setIsExecuteProposalModalOpen(true);
+    } else {
+      alert("Cannot execute proposal.");
+    }
+  };
+
+  const getProposalDetails = async () => {
+    const proposal = demoProposalData[0];
+    if (proposal) {
+      setMaintainers(proposal.maintainers);
+      setVoteStatus(proposal.voteStatus);
+      setProposalStatus(proposal.status);
+      setEndDate(proposal.endDate);
+      const description = await fetchProposalFromIPFS(proposal.ipfsLink);
+      setDescription(description);
+      const outcome = await fetchOutcomeDataFromIPFS(proposal.ipfsLink);
+      setOutcome(outcome);
+    } else {
+      alert("Proposal not found");
+    }
+  };
+
+  useEffect(() => {
+    getProposalDetails();
+  }, [id]);
+
+  return (
+    <div>
+      <ProposalPageTitle
+        id={id}
+        title="Bounty of issue: integrate DAO system to Tansu - $3000"
+        submitVote={() => openVotingModal()}
+        executeProposal={() => openExecuteProposalModal()}
+        status={proposalStatus}
+        maintainers={maintainers}
+      />
+      <div className="flex flex-col gap-3 sm:gap-5 md:gap-7">
+        <ProposalStatusSection status={proposalStatus} endDate={endDate} />
+        <VoteStatusBar
+          interest={50}
+          conflict={30}
+          abstain={20}
+          onClick={(voteType) => openVotersModal(voteType)}
+        />
+        <ProposalDetail
+          description={description}
+          outcome={outcome}
+          status={proposalStatus}
+        />
+      </div>
+      {isVotersModalOpen && voteStatus && (
+        <VotersModal
+          NQGScore={4.5}
+          voteData={voteType ? voteStatus[voteType] : null}
+          onClose={() => setIsVotersModalOpen(false)}
+        />
+      )}
+      {isVotingModalOpen && (
+        <VotingModal onClose={() => setIsVotingModalOpen(false)} />
+      )}
+      {isExecuteProposalModalOpen && (
+        <ExecuteProposalModal
+          xdr={outcome?.approved.xdr ?? ""}
+          onClose={() => setIsExecuteProposalModalOpen(false)}
+        />
+      )}
+    </div>
+  );
```

Comment:
I suppose it's a left over as you should use the mocked data. All that should be abstracted so that once you call the contract you just have to replace the mock data with the actually contract calls.